### PR TITLE
4 feature 4 search stock

### PR DESCRIPTION
**Pull request by noel-toms**
Added tests and finished all the features. Also have updated some of the APIInterface functions for my use case. This is so I can get historical data for a specified range. As the only real input, this use case takes is a stock symbol I have added tests to test both a correct and incorrect symbol.

### CODE REVIEW

**Comment by eladperlman** [comment]
One of your tests failed

**Comment by noel-toms** [comment]
it's strange as the test that fails is structured exactly as the other test and runs fine on my computer.  It says there's an assertion error however it's not showing on IntelliJ.

**Comment by noel-toms** [comment]
> One of your tests failed

I've fixed my test and it passes the auto-grading. The reason it's still failing is because of another test that's not part of my use case. The failing test is this:

  RegisterPresenterTest > testNoIssue() FAILED
      java.lang.RuntimeException at RegisterPresenterTest.java:44
          Caused by: java.sql.SQLException at RegisterPresenterTest.java:44

**Comment by noel-toms** [comment]
@inaam10-tech  I believe this is your test. Please look at the issue outlined in my previous comment

**Review by inaam10-tech — DISMISSED** [review]
Seems good.  I'll see why my test failed.

**Review by benmarlow — APPROVED** [review]
Apart from that test failing which isn't even in your branch, it all looks good to me. Good work.

**Comment by Peaceful36** [comment]
> Seems good. I'll see why my test failed.

Just checked, all of your tests pass now.

File: src/main/java/SearchStockUseCase/ViewStockResponseModel.java [hunk 1]
```diff
@@ -0,0 +1,4 @@
+package SearchStockUseCase;
+
+public class ViewStockResponseModel {
```
Comment (Peaceful36) [inline src/main/java/SearchStockUseCase/ViewStockResponseModel.java:3]: I think you forgot to implement this class

**Review by Peaceful36 — CHANGES_REQUESTED** [review]

**Review by Spaghetti-12 — APPROVED** [review]
Looks good!

File: src/main/java/SearchStockUseCase/ViewStockResponseModel.java (hunk 1 above)
Comment (noel-toms) [reply src/main/java/SearchStockUseCase/ViewStockResponseModel.java:3]: Good catch I removed the unused files now

**Pull request merged**